src/VisualStudio/IntegrationTest/New.IntegrationTests/AbstractIntegrationTest.cs

```diff
@@ -35,6 +38,32 @@ static AbstractIntegrationTest()
             RuntimeHelpers.RunModuleConstructor(typeof(TestBase).Module.ModuleHandle);
             TestTraceListener.Install();
 
+            DataCollectionService.RegisterCustomLogger(
+                static fileName =>
+                {
+                    if (s_editorPackage is null)
+                        return;
+
+                    var folder = s_editorPackage.UserLocalDataPath;
```

Comment:
Why are we asking the editor package for this versus just calling one of the other APIs for this same path?

Reply:
Would love to learn about APIs that might exist for getting the right path here. I wrote a rather brute force approach in Razor to do it, which I've always wondered when it might break: https://github.com/dotnet/razor/blob/main/src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/VisualStudioLogging.cs#L134

Reply:
➡️ I now obtain the path from `IVsComponentModelHost`